modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -660,6 +662,27 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
     }
 
     /**
+     * If the experiment is a rapid experiment, it checks whether the user cap has already been achieved.
+     * If yes, then it sets the experiment state to paused, refreshes the experiment metadata cache
+     * and returns false i.e. assignment is not allowed in that case.
+     */
+    private boolean checkUserCapForRapidExperiment(Experiment experiment) {
```

Comment:
@bargenilesh AFAIK, the PUT assignment flow is for admin use to update assignments only. Do we want to really enforce this check here?